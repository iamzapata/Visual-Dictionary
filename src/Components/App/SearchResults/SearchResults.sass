.SearchResults
  position: relative

  ul
    width: 100%
    display: flex
    align-items: center
    justify-content: center
    flex-direction: column

  li:last-of-type hr
    display: none

  li
    width: 100%


