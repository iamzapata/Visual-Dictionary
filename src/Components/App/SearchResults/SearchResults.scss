.SearchResults {
  position: relative;

  .SearchResults__Definitions {
    ul {
      width: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
    }

    li:last-of-type hr {
      display: none;
    }
    li {
      width: 100%;
    }
  }

  .SearchResults__Images {
    margin: 20px 0 50px 0;
    .slick-track {
      display: flex;
      align-items: center;

      img {
        width: auto;
        height: auto;
      }

      button.slick-arrow::before {
        color: black !important;
      }
    }
  }
}
