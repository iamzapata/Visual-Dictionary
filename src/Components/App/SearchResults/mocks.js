import shortId from "shortid"

const entrySample = () => ({
  id: shortId.generate(),
  text: "privacy",
  language: "es",
  lexicalCategory: "Noun",
  pronunciations: [
    {
      id: shortId.generate(),
      audioFile: "http://audio.oxforddictionaries.com/en/mp3/privacy_gb_1.mp3",
      phoneticSpelling: "prɪvəsi",
      dialects: ["British English"]
    },
    {
      id: shortId.generate(),
      phoneticSpelling: "prʌɪvəsi",
      dialects: ["British English"]
    }
  ],
  entries: [
    {
      id: shortId.generate(),
      etymologies: ["Lorem ipsum dolor sit amet, consectetuer adipiscing elit"],
      senses: []
    }
  ]
})

export default entrySample
