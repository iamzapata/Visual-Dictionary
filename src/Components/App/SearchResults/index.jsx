import React, { PureComponent } from "react"
import { arrayOf, shape, string } from "prop-types"
import Slider from "react-slick"
import LexicalEntry from "./Components/LexicalEntry"
import "./SearchResults.sass"

const sliderSettings = {
  className: "",
  center: true,
  dots: true,
  infinite: true,
  slidesToShow: 4,
  slidesToScroll: 1
}

const renderResultItem = entry => {
  return (
    <li key={entry.id}>
      <LexicalEntry entry={entry} />
      <hr />
    </li>
  )
}

class SearchResults extends PureComponent {
  render() {
    const { results, imageResults } = this.props

    return (
      <div className="SearchResults">
        <div className="SearchResults__Images">
          <Slider {...sliderSettings}>
            {imageResults.map(img => (
              <a
                href={img.link}
                target="_blank"
                key={img.link}
                rel="noopener noreferrer"
              >
                <img src={img.thumbnailLink} alt="" />
              </a>
            ))}
          </Slider>
        </div>
        <ul className="SearchResults__Definitions">
          {results.map(entry => renderResultItem(entry))}
        </ul>
      </div>
    )
  }
}

SearchResults.propTypes = {
  results: arrayOf(
    shape({
      id: string,
      text: string,
      language: string,
      lexicalCategory: string,
      entries: arrayOf(shape({})),
      pronunciations: arrayOf(shape({}))
    })
  ).isRequired,

  imageResults: arrayOf(
    shape({
      link: string,
      thumbnailLink: string
    })
  ).isRequired
}

export default SearchResults
