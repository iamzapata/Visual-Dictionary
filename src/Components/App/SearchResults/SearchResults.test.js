import React from "react"
import SearchResults from "./index.jsx"
import { renderIntoDocument } from "react-testing-library"

describe("<SearchResults />", () => {
  it("Should render self", () => {
    const { container } = renderIntoDocument(<SearchResults />)

    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("Should have a SearchResults className", () => {
    const { container } = renderIntoDocument(<SearchResults />)
    const classNameSelection = container.querySelectorAll(".SearchResults")

    expect(classNameSelection.length).toEqual(1)
  })

  it("Should have an ul element", () => {
    const { container } = renderIntoDocument(<SearchResults />)

    const unorderdList = container.querySelector("ul")

    expect(unorderdList).toBeTruthy()
  })

})
