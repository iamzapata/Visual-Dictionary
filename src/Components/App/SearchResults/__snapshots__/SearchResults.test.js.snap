// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> Should render self 1`] = `
<div
  class="SearchResults"
>
  <div
    class="SearchResults__Images"
  >
    <div
      class="slick-slider slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                href="link1"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt=""
                />
              </a>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                href="link2"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt=""
                />
              </a>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                href="link3"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt=""
                />
              </a>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-cloned"
            data-index="3"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                href="link1"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt=""
                />
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="4"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                href="link2"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt=""
                />
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="5"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                href="link3"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt=""
                />
              </a>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
  <ul
    class="SearchResults__Definitions"
  >
    <li>
      <div
        class="LexicalEntry"
      >
        <h4
          class="title is-4"
        >
          privacy
        </h4>
        <h5
          class="LexicalCategory is-italic title is-5 has-text-weight-light"
        >
          Noun
        </h5>
        <div
          class="Pronunciations"
        >
          <span>
            /
            <span>
              prɪvəsi
            </span>
            /
            <p>
              <span
                class="Dialects"
              >
                <span>
                  British English
                </span>
              </span>
              <button
                class="PlayAudioButton"
              >
                ►
              </button>
            </p>
          </span>
          <span>
            /
            <span>
              prʌɪvəsi
            </span>
            /
            <p>
              <span
                class="Dialects"
              >
                <span>
                  British English
                </span>
              </span>
            </p>
          </span>
        </div>
        <div
          class="Senses"
        >
          <ul />
        </div>
        <div
          class="Etymologies"
        >
          <h6
            class="title is-6"
          >
            Origin:
          </h6>
          <ul>
            <li>
              Lorem ipsum dolor sit amet, consectetuer adipiscing elit
            </li>
          </ul>
        </div>
      </div>
      <hr />
    </li>
  </ul>
</div>
`;
